### DISCUSSION TITLE
How to set a timeout for fetch?

### DISCUSSION THREAD

**Discussion by zaru** [vercel/next.js#72443] [Help]
### Summary

When I make a request to an HTTP endpoint using fetch, it times out after 300 seconds.

This seems to be the default value of undici inside Node.js.
Undici itself has options where you can specify the timeout.
https://github.com/nodejs/undici/blob/ae91a25e4771778cac41f2fb2b0c0f1f60024c44/docs/docs/api/Dispatcher.md

However, I don’t know how to specify this from Next.js. Do you have any good ideas?

**Comment by icyJoseph** [comment]
Yeah, I think, in general this is a bit hard to do in Node.js itself, or rather I mean, without involving undici somehow, https://github.com/nodejs/undici/issues/1373 

Looking at this piece of code:

```ts
import { fetch, Agent } from 'undici'
import { createServer } from 'http'
import { once } from 'events'

const agent = new Agent({
  headersTimeout: 100
})

const server = createServer(() => {}).listen(0)

await once(server, 'listening')

try {
  await fetch(`http://localhost:${server.address().port}`, {
    dispatcher: agent
  })
} catch (e) {
  if (e.message === 'fetch failed') {
    console.log(e.cause)
  }
}
```

It'd look as if you'd need to create an agent, and then pass that to each fetch call you make... not very practical

Some others suggest overriding a global symbol, but it is not clear how to do that either.

**Comment by syofyanzuhad** [comment]
this solution solve my problem: https://stackoverflow.com/a/77005381/11871997

```
TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11730:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: ConnectTimeoutError: Connect Timeout Error
      at onConnectTimeout (node:internal/deps/undici/undici:6869:28)
      at node:internal/deps/undici/undici:6825:50
      at Immediate._onImmediate (node:internal/deps/undici/undici:6857:13)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'UND_ERR_CONNECT_TIMEOUT'
  }
}
```
